pyvista/plotting/plotter.py

```
@@ -4462,7 +4462,7 @@ def add_volume(  # noqa: PLR0917
             #       Also, place all data on the nodes as issues arise when
             #       volume rendering on the cells.
             volume = volume.cell_data_to_point_data()
-        assert isinstance(volume, (pv.DataSet, pv.MultiBlock))
+        assert isinstance(volume, (pv.DataSet, pv.MultiBlock))  # noqa: S101
```

Let's raise an error here!

---

this check is done a few lines earlier with `_validation.check_instance` on the `not is_pyvista_dataset(volume)` branch:
https://github.com/pyvista/pyvista/blob/8cfe9b69637ad140376768526d208f18d967cc7b/pyvista/plotting/plotter.py#L4450-L4459

So we already raise a TypeError here. But I think the issue is `mypy` doesn't realize we have already narrowed the type, so it thinks the `volume` variable can still be a non-mesh type. If we refactor a bit to make `mypy` happy, we can remove the need for this assertion. But `volume` is a heavily used variable so might not be so straightforward